### DISCUSSION TITLE
Convert DWG file

### DISCUSSION THREAD

**Discussion by a-nejjari** [DomCR/ACadSharp#154] [Q&A]
Hi everyone,

Is it possible to convert DWG to SVG? If yes: Can I translate each block seperatly? 

Thanks in advance

**Comment by DomCR** [comment]
Hi @a-nejjari,

I'm sorry but this feature is not implemented.

I'll consider the future implementation but it will take a while.

**Reply by a-nejjari** [reply]
Okay thank you for your answer.

**Comment by mme1950** [comment]
We are currently implementing DWG-to-SVG converter. We can already convert a lot of entities.
Converting block records into SVG groups and converting subgrouping of dynamic blocks into SVG (sub)groups will be an essential feature.
see also #161.

**Reply by a-nejjari** [reply]
Hello @mme1950,
Thank you for your response. Where can I find an example or explanation?

**Comment by mme1950** [comment]
Hi @a-nejjari Thank you for your interst.

We will set up a repository in GitHub soon. But our solution is still in a prototype state. 

Our converter can read a DWG file and convert it to SVG. The converted code can be edited in a Scintilla text box. The result is displayed in a browser control.
We are interested in the SVG-groups created from DWG-Blocks. Thus the SCG converter focuses on the conversion of the block structure, in particular dynamic blocs, rather than just converting a drawing.

The converter supports many entities such as Arc, Circle, Dimensions, Ellipse, Hatch, Insert, Leader, Line, Lwpolyline, MText, Multileader, Spline, TextEntity. But many details such as colours or line types are not yet handled.

The solution is based on ACadSharp. We are currently working on two extensions to ACadSharp that shall be integrated into the main project:
* Reading Multileader entities.
* Reading dynamic blocks.

Both features are used by our converter.

**Comment by mme1950** [comment]
Hi @a-nejjari, see https://github.com/nanoLogika.
